### ISSUE TITLE
Hard-coded Date should be created in UTC

### ISSUE THREAD

**Issue by rightaway** [eslint/eslint#13069]
**Please describe what the rule should do:**

It could be dangerous to just use `new Date(args)` with arguments for year, month, day because depending on what time zone you're running it from you'll get different values and it's not always the expected behavior. If you run it as `new Date(Date.UTC(args))` you'll get consistent date value always.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
new Date(2020,2,21) // will fail
new Date(Date.UTC(2020,2,21)) // will pass

```

**Why should this rule be included in ESLint (instead of a plugin)?**
It's very important to use Date correctly.

**Comment by kaicataldo** [comment]
This feels like a good candidate for a custom rule to me.

**Comment by rightaway** [comment]
Date is native javascript so having this as a core rule will help developers avoid nasty surprises. There's probably developers who don't know Date works like this, but seeing this eslint rule will let them know and warn them when they're doing it wrong.

**Comment by kaicataldo** [comment]
Agreed that it's valuable. At this point we have a really large number of rules and have to set a very high bar for inclusion of new rules. Let's see what other team members think.
